testing/test_pytest_html.py

```diff
@@ -894,9 +887,6 @@ def test_css_invalid(self, testdir, recwarn):
         assert "No such file or directory: 'style.css'" in result.stderr.str()
 
     def test_css_invalid_no_html(self, testdir):
```

Comment:
If you mark the tests to filter warnings:

```python
@pytest.mark.filterwarnings("ignore:.*TerminalReporter.writer.*")
```

You might not need `remove_deprecation_from_recwarn` in the end?

Reply:
Tried it and it didn't work @nicoddemus 

Maybe I'm doing something wrong?

```python
    @pytest.mark.filterwarnings("ignore:.*TerminalReporter.writer.*")
    @pytest.mark.parametrize("colors", [(["red"]), (["green", "blue"])])
    def test_css(self, testdir, recwarn, colors):
        testdir.makepyfile("def test_pass(): pass")
        css = {}
        cssargs = []
        for color in colors:
            style = f"* {{color: {color}}}"
            path = testdir.makefile(".css", **{color: style})
            css[color] = {"style": style, "path": path}
            cssargs.extend(["--css", path])
        result, html = run(testdir, "report.html", "--self-contained-html", *cssargs)
        assert result.ret == 0
        assert len(recwarn) == 0
        for k, v in css.items():
            assert str(v["path"]) in html
            assert v["style"] in html
```

Reply:
Yeah unfortunately the mark is considered only for the test run, not the setup (which is where the warning is issued).